airbyte-integrations/connectors/source-stripe/source_stripe/availability_strategy.py

```diff
@@ -0,0 +1,31 @@
+import logging
+from typing import Optional, Tuple
+
+from airbyte_cdk.sources import Source
+from airbyte_cdk.sources.streams import Stream
+from airbyte_cdk.sources.streams.http.availability_strategy import HttpAvailabilityStrategy
+
+
+class StripeSubStreamAvailabilityStrategy(HttpAvailabilityStrategy):
+    def check_availability(self, stream: Stream, logger: logging.Logger, source: Optional[Source]) -> Tuple[bool, Optional[str]]:
+        """Traverse through all the parents of a given stream and run availability strategy on each of them"""
+        try:
+            current_stream, parent_stream = stream, getattr(stream, "parent")
+        except AttributeError:
+            return super().check_availability(stream, logger, source)
```

Comment:
Just wondering, why try/except here, while inside the loop you use `getattr`? Because we assume if we are using the strategy that this stream will have a parent and if we don't hit it it's the exception?

Reply:
Cool, thank you for clarifying. I do think its reasonable